oolab/src/main/java/agh/ics/oop/RectangularMap.java

```diff
@@ -0,0 +1,57 @@
+package agh.ics.oop;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RectangularMap implements IWorldMap{
+
+    private List<Animal> animals = new ArrayList<>();
+    private static final Vector2d LOWER_BOUND = new Vector2d(0,0);
+    private static Vector2d UPPER_BOUND; /* Yup, I know it's not a constant value,
+    but I will still use it like that, because it's convenient */
+
+    RectangularMap(int width, int height){
+        UPPER_BOUND = new Vector2d(width-1, height-1);
+    }
+
+    @Override
+    public boolean canMoveTo(Vector2d dest){
+        if(!(LOWER_BOUND.precedes(dest))) return false;
+        if(!(UPPER_BOUND.follows(dest))) return false;
+        if(isOccupied(dest)) return false;
+        return objectAt(dest) == null;
```

Comment:
jeśli sprawdzimy `isOccupied()` to raczej nie trzeba już `objectAt()`? + można te warunki złączyć koniunkcją `&&` i pozbyć się ifa